File: ios/NativeMapsModule.swift
import ExpoModulesCore
import MapKit

public class NativeMapsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeMaps")

    View(NativeMapsView.self) {
      Prop("initialRegion") { (view: NativeMapsView, region: [String: Double]) in
        guard let lat = region["latitude"],
              let lng = region["longitude"] else { return }
        
        let zoom = region["zoom"] ?? 12.0
        guard zoom > 0 && zoom.isFinite else { return }
        
        let center = CLLocationCoordinate2D(latitude: lat, longitude: lng)
        let deltaFromZoom = 360.0 / pow(2.0, zoom)
        let span = MKCoordinateSpan(latitudeDelta: deltaFromZoom, longitudeDelta: deltaFromZoom)
        let region = MKCoordinateRegion(center: center, span: span)
        view.mapView.setRegion(region, animated: false)
      }
      
      Prop("showsUserLocation") { (view: NativeMapsView, showsUserLocation: Bool) in
        view.mapView.showsUserLocation = showsUserLocation
      }
      
      Prop("mapType") { (view: NativeMapsView, mapType: String) in
        switch mapType {
        case "satellite":
          view.mapView.mapType = .satellite
        case "hybrid":
          view.mapView.mapType = .hybrid
        default:
          view.mapView.mapType = .standard
        }
      }
      
      Events("onMapPress")
    }
  }
}

File: android/src/main/java/expo/modules/nativemaps/NativeMapsModule.kt
package expo.modules.nativemaps

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Production-ready Native Maps module for Android
 * Provides react-native-maps compatible API
 */
class NativeMapsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeMaps")

    View(NativeMapsView::class) {
      // Props following react-native-maps API
      Prop("initialRegion") { view: NativeMapsView, region: Map<String, Double> ->
        val lat = region["latitude"] ?: return@Prop
        val lng = region["longitude"] ?: return@Prop
        val zoom = region["zoom"] ?: 12.0
        view.setInitialRegion(lat, lng, zoom)
      }
      
      Prop("showsUserLocation") { view: NativeMapsView, shows: Boolean ->
        view.setShowsUserLocation(shows)
      }
      
      Prop("mapType") { view: NativeMapsView, type: String ->
        view.setMapType(type)
      }
      
      // Event following react-native-maps convention
      Events("onMapPress")
    }
  }
}